### PR TITLE
Replace Prop drilling with useContext pattern for access to state

### PR DESCRIPTION
**Pull request by aymanhab**
This PR refactors state management for the viewer components by making them share a state using the "useContext" /"ContextProvider" pattern so we don't pass loads of properties down the tree of components. This also fixes the issue where graphical and tree selection are out of sync.

### CODE REVIEW

**Comment by aymanhab** [comment]
@AlbertoCasasOrtiz for your review/feedback. Thank you

**Comment by AlbertoCasasOrtiz** [comment]
Hi @aymanhab, selecting in the tree works perfectly.

I found three issues:

1. If the first thing you do is go to the viewer (without selecting anything in the model showcase before), it throws the following error:

```
ERROR
Cannot read properties of null (reading 'name')
TypeError: Cannot read properties of null (reading 'name')
    at new SceneTreeModel (http://localhost:3000/static/js/bundle.js:3596:42)
    at SceneTreeView (http://localhost:3000/static/js/bundle.js:1068:27)
    at http://localhost:3000/static/js/bundle.js:58046:14
    at http://localhost:3000/static/js/bundle.js:58377:19
    at trackDerivedFunction (http://localhost:3000/static/js/bundle.js:61084:18)
    at Reaction.track (http://localhost:3000/static/js/bundle.js:61599:18)
    at useObserver (http://localhost:3000/static/js/bundle.js:58375:16)
    at observerComponent (http://localhost:3000/static/js/bundle.js:58045:69)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:80034:22)
    at updateFunctionComponent (http://localhost:3000/static/js/bundle.js:82916:24)
```

This error is only shown locally, but not in the deploy visualization (where it just stays blank, I guess because debug is not activated in the deploy preview).

2. When opening the left menu in the viewer, the rest of the interface is displaced as the horizontal scrollbar appears.

3. This I think is not an error, and could be interesting keeping it. When you modify something in the viewer (for example, move or rotate, or select any mesh), and then you go to the model showcase, all of the changes are shown in the showcase previsualuzation.

**Comment by aymanhab** [comment]
Thanks for the review @AlbertoCasasOrtiz I fixed issue 1 in the latest commit.
Issue 3 is a bit more complicated but I'm working on a fix for it, your call if we should hold on merging for the fix.
I'd leave issue 2 for discussion later as we evolve the UI for example the 3d window could resize if the intention is to make the drawer open regularly as opposed to open the drawer, change settings then close. Up for discussion. This may differ between phone/tablet/desktop

**Comment by AlbertoCasasOrtiz** [comment]
@aymanhab Just tested issue 1 in the deploy preview and works perfectly now.

For issue 2, I agree, it would be better to discuss the behaviour on different devices. Maybe we could even let it open by default in desktop as other apps do (blender, unity... Etc). We can discuss later and maybe ask for opinions in a future demo meeting.

For issue 3, I think we could work on it in another PR, so we can merge this one.

**Comment by aymanhab** [comment]
Awesome, thanks @AlbertoCasasOrtiz will merge an open issues to address 2 & 3

**Comment by aymanhab** [comment]
PR still needs approval though 😃

**Review by AlbertoCasasOrtiz — APPROVED** [review]
I just reviewed all of the files and everything seems fine.

**Comment by aymanhab** [comment]
Thanks @AlbertoCasasOrtiz

**Pull request merged**
